ddsketch/ddsketch.go

```diff
@@ -255,6 +255,9 @@ func FromProtoWithStoreProvider(pb *sketchpb.DDSketch, storeProvider store.Provi
 // accuracy, but it avoids artefacts like empty bins that make the histograms look bad.
 // scaleFactor allows to scale out / in all values. (changing units for eg)
 func (s *DDSketch) ChangeMapping(newMapping mapping.IndexMapping, positiveStore store.Store, negativeStore store.Store, scaleFactor float64) *DDSketch {
+	if scaleFactor == 1 && s.IndexMapping.Equals(newMapping) {
+		return s
```

Comment:
For consistency with the other path, I think we should return a copy (btw, the documentation explicitly says `it doesn't change s but returns a newly created sketch.`).

```suggestion
		return s.Copy()
```